### PR TITLE
Clarify the type restrictions documentation

### PR DESCRIPTION
**Pull request by GloverDonovan**
This pull request adds a section demonstrating how Crystal handles type restricted variables inside the actual methods.

I want this section to be added, but the example could be much better than it is now. I'm not sure what to put there.

### CODE REVIEW

File: syntax_and_semantics/type_restrictions.md [hunk 1]
```diff
@@ -69,6 +69,15 @@ restricted_add Six.new, 10
 
 Refer to the [type grammar](type_grammar.html) for the notation used in type restrictions.
 
+Note that type restrictions do not apply to the variables inside the actual methods. This lets you do useful things with them.
```
Comment (RX14) [inline syntax_and_semantics/type_restrictions.md]: >This lets you do useful things with them

It's a bit wishy washy.

Comment (GloverDonovan) [reply syntax_and_semantics/type_restrictions.md]: How about: `This lets you use the same variable name for different types`?

Comment (straight-shoota) [reply syntax_and_semantics/type_restrictions.md]: Just leave that "useful" sentence out.

Comment (GloverDonovan) [reply syntax_and_semantics/type_restrictions.md]: Done

**Comment by mverzilli** [comment]
Thank you! This is a usual source of confusion about the language, it's nice to have it in a place we can link to :)

**Pull request merged**